ledger/src/shred/merkle_tree/fec_set_root.rs

```diff
@@ -0,0 +1,40 @@
+//! Module to define FecSetRoot which represents the merkle root of a single FEC set (Slice).
+
+use {
+    serde::{Deserialize, Serialize},
+    solana_hash::{Hash, HASH_BYTES},
+};
+
+/// Using the new type pattern, define a type to represent the merkle root of a single FEC set (Slice).
+#[derive(
+    Copy, Clone, Default, Debug, Hash, PartialEq, Eq, PartialOrd, Ord, Serialize, Deserialize,
+)]
+#[repr(transparent)]
+pub struct FecSetRoot([u8; HASH_BYTES]);
+
+impl FecSetRoot {
+    /// Constructs a new [`FecSetRoot`] from a [`Hash`].
+    ///
+    /// Visbility is restricted just to the [`merkle_tree`] module.
```

Comment:
The restricted visibility is self evident from the code. The question is why/if we want to restrict visibility here.